op_crates/fetch/26_fetch.js

```diff
@@ -941,6 +949,122 @@
     return m;
   }
 
+  // Headers should be able to convert values to HeadersInit,
+  // internally work with Headers directly
+  const headersInitConverter = (v) => new Headers(v);
+
+  // https://fetch.spec.whatwg.org/#bodyinit
+  const bodyInitConverter = () => {};
+
+  // https://w3c.github.io/webappsec-referrer-policy/#enumdef-referrerpolicy
+  const referrerPolicyConverter = () => {};
+
+  // https://fetch.spec.whatwg.org/#requestmode
+  const requestModeConverter = createEnumConverter(
+    "RequestMode",
+    ["navigate", "same-origin", "no-cors", "cors"],
+  );
+  // createEnumConverter
+  // createNullableConverter,
+  // https://fetch.spec.whatwg.org/#requestcredentials
+  const requestCredentialsConverter = createEnumConverter(
+    "RequestCredentials",
+    ["omit", "same-origin", "include"],
+  );
+
+  // https://fetch.spec.whatwg.org/#requestcache
+  const requestCacheConverter = createEnumConverter(
+    "RequestCache",
+    [
+      "default",
+      "no-store",
+      "reload",
+      "no-cache",
+      "force-cache",
+      "only-if-cached",
+    ],
+  );
+
+  // https://fetch.spec.whatwg.org/#requestredirect
+  const requestRedirectConverter = createEnumConverter(
+    "RequestRedirect",
+    [
+      "follow",
+      "error",
+      "manual",
+    ],
+  );
+
+  const abortSignalConverter = () => {};
+
+  // https://fetch.spec.whatwg.org/#requestinit
+  const requestInitConverter = createDictionaryConverter("RequestInit", [
+    {
+      converter: webidl.converter.ByteString,
+      key: "method",
+    },
+    {
+      converter: headersInitConverter,
+      key: "headers",
+    },
+    {
+      // BodyInit?
+      converter: createNullableConverter(bodyInitConverter),
+      key: "body",
+    },
+    {
+      converter: webidl.converter.USVString,
+      key: "referrer",
+    },
+    {
+      converter: referrerPolicyConverter,
+      key: "referrerPolicy",
+    },
+    {
+      converter: requestModeConverter,
+      key: "mode",
+    },
+    {
+      converter: requestCredentialsConverter,
+      key: "credentials",
+    },
+    {
+      converter: requestCacheConverter,
+      key: "cache",
+    },
+    {
+      converter: requestRedirectConverter,
+      key: "redirect",
+    },
+    {
+      converter: webidl.converter.DOMString,
+      key: "integrity",
+    },
+    {
+      converter: webidl.converter.boolean,
+      key: "keepalive",
+    },
+    {
+      // AbortSignal?
+      converter: createNullableConverter(abortSignalConverter),
+      key: "signal",
```

Comment:
It seems like some of these are unused in Deno's implementation, ex: `signal`.
I'll leave implementing them to another PR, as I'd only like to make `Request`, `Response`, and `fetch` use proper type conversions.

I'm starting to think that I might not even attempt to de-dup `fetch` and use `Request` in it's implementation, would anyone consider that out of scope for this change?

As for type checking / converting some of these, I'm not sure if Deno even implements `AbortSignal`, does it?

Reply:
On second thought, passing anything that isn't implemented should throw a "not implemented" error.

Reply:
No any extra priorities that are but used should be silently ignored.

Reply:
> ...that are but used...

It wasn't clear what you were intending to say, did you miss a word?

Reply:
Im sorry, phone keyboard. Here we go again:

Extra properties that are not implemented by us at the moment should be silently ignored.

Reply:
Okay, got it.